File: EASY/680.Valid Palindrome/Solution.py
class Solution:
    def isplaindrom(self, s, l, r):
        while l<r:
            if s[l]!=s[r]:
                return False
            l += 1
            r -= 1
        return True
    def validPalindrome(self, s: str) -> bool:
        n = len(s)
        l = 0 
        r = n-1
        while l<r:
            if s[l]!=s[r]:
                return self.isplaindrom(s, l+1, r) or self.isplaindrom(s, l, r-1)
            l = l+1
            r = r-1
        return True

File: EASY/680.Valid Palindrome/Solution.cpp
class Solution {
public:
    bool validPalindrome(string s) {
          int j = s.size()-1;
        for(int i = 0; i<j;i++,j--){
            if(s[i]!=s[j]){
               int i1 = i+1; int j1 = j;
               while(i1<j1 and s[i1]==s[j1]){
                i1++; j1--;
               }
               if(i1>=j1) return true;
               int i2 = i; int j2 = j-1;
               while(i2<j2 and s[i2] == s[j2]){
                j2--; i2++;
               }
               if(i2>=j2) return true;

               return false;
            }
        }
        return true;
    }
};
